src/rp2_common/hardware_dma/include/hardware/dma.h

```diff
@@ -891,6 +891,18 @@ static inline uint dma_get_timer_dreq(uint timer_num) {
     return DREQ_DMA_TIMER0 + timer_num;
 }
 
+/*! \brief Performs DMA channel cleanup after use
+ *  \ingroup hardware_dma
+ *
+ * \param channel DMA channel
+ */
+static inline void dma_channel_cleanup(uint channel) {
+    dma_channel_set_irq0_enabled(channel, false);
+    dma_channel_set_irq1_enabled(channel, false);
+    dma_channel_abort(channel);
+    hw_clear_bits(&dma_hw->intr, 1u << channel);
```

Comment:
should just be a regular write of `dma_hw->intr`; it is a WC field

Reply:
Apologies - that was dumb. I think I fixed it now.